src/snakemake/io/resolve_file_format.py

```
@@ -0,0 +1,126 @@
+
+
+def resolve_file_format(
+    file: str,
+) -> Tuple[Callable[[str], Any], Callable[[Any, str], None]]:
+    suffix = file.rsplit(".", 1)[-1]
+    open_: Callable[[str, str], Any]
+    match suffix:
+        case "gz":
+            import gzip
+
+            open_ = lambda f, mode: gzip.open(f, f"{mode}b")
```

Ruff (0.15.7)

[error] Do not assign a `lambda` expression, use a `def`

Rewrite `open_` as a `def`

(E731)